paimon-core/src/main/java/org/apache/paimon/Changelog.java

```diff
@@ -0,0 +1,132 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.paimon;
+
+import org.apache.paimon.fs.FileIO;
+import org.apache.paimon.fs.Path;
+import org.apache.paimon.utils.JsonSerdeUtil;
+
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonCreator;
+import org.apache.paimon.shade.jackson2.com.fasterxml.jackson.annotation.JsonProperty;
+
+import javax.annotation.Nullable;
+
+import java.io.IOException;
+import java.util.Map;
+
+/**
+ * The metadata of changelog. It generates from the snapshot file during expiration. So that the
+ * changelog of the table can outlive the snapshot's lifecycle. A table's changelog can come from
+ * two source:
+ * <li>The changelog file. Eg: from the changelog-producer = 'input'
+ * <li>The delta files in the APPEND commits when the changelog-producer = 'none'
+ */
+public class Changelog extends Snapshot {
+
+    private static final int CURRENT_VERSION = 3;
```

Comment:
The current version is set to 3, equal to the version of snapshot. Otherwise such as `1`, when pass the `Changelog` as `Snapshot` to `org.apache.paimon.operation.AbstractFileStoreScan#readManifests()` it will be recognize as a table_store_02_version.

Reply:
Set version to 1 now. Refactor logic in `AbstractFileStoreScan#readManifests()`